hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -1833,7 +1844,11 @@ protected void stopServiceThreads() {
   }
 
   private void createProcedureExecutor() throws IOException {
-    MasterProcedureEnv procEnv = new MasterProcedureEnv(this);
+    final String procedureDispatcherClassName =
+      conf.get(HBASE_MASTER_RSPROC_DISPATCHER_CLASS, DEFAULT_HBASE_MASTER_RSPROC_DISPATCHER_CLASS);
```

Comment:
Is this for extention of RSProcedureDispatcher or can there be a new implementation of RemoteProcedureDispatcher ?

Reply:
This allows only extension of RSProcedureDispatcher, mainly to be used for testing purpose.